sdk/sdk.go

```diff
@@ -73,7 +73,7 @@ func (agent *Agent) Shutdown() error {
 
 // AddReport adds a new usage report.
 func (agent *Agent) AddReport(report metrics.MetricReport) error {
-	return agent.input.AddReport(report)
+	return agent.input.AddReport(report.Copy())
```

Comment:
do we need to update any of the AddReport(...) usages in 

https://github.com/GoogleCloudPlatform/ubbagent/blob/a68c92c74f980482ffb1b313b38d3d61e640818c/pipeline/inputs/inputs.go

Reply:
No, this change is only required at the sdk and heartbeat boundaries